### DISCUSSION TITLE
Question with Queues and Processors

### DISCUSSION THREAD

**Discussion by nivekithan** [OptimalBits/bull#2545] [Q&A]
I have a question regarding Queues and Processors. 

So lets say we two services. One is the main backend service and another is Job Processor service. Backend service listens to requests and creates the job and puts into the queue using `queue.add` method. 

Whereas Job Processor service intializes both `queue` and `processors` for thoses queues. 

In this case lets say we got a request which creates 100 Jobs,  is there any gurantee that all Jobs will be consumed only by `Job Processor` service ( if processor is not facing any other kind of problem) and that `Backend` service will not hold onto certain jobs waiting for `Processors` to come online in its instance ?

Thanks

**Comment by manast** [comment]
Not sure I understand your question, but only the service implementing the processor will work on the job and the service adding the jobs do not need to wait for the job to be processed.
